### PR TITLE
MTKA 1487 CRUD add translation strings to validate field functions

### PR DESCRIPTION
**Pull request by anthonyburchell**
## Description

This PR adds translations for field validation error messages.

### CODE REVIEW

**Review by nickcernis — DISMISSED** [review]

File: includes/api/validation-functions.php [hunk 1]
```diff
@@ -252,7 +254,9 @@ function validate_number( $value, $message = 'Value must be a valid number' ): v
  *
  * @return void
  */
-function validate_date( $value, $message = 'Value must be of format YYYY-MM-DD' ): void {
+function validate_date( $value, $message = null ): void {
```
Comment (mindctrl) [inline includes/api/validation-functions.php:257]:
```suggestion
function validate_date( $value, string $message ): void {
```

File: includes/api/validation-functions.php [hunk 2]
```diff
@@ -342,7 +354,9 @@ function validate_array_key_exists( $key, array $array, $message = 'The key is r
  *
  * @return void
  */
-function validate_not_empty( $value, string $message = 'The field cannot be empty' ): void {
+function validate_not_empty( $value, string $message = null ): void {
```
Comment (mindctrl) [inline includes/api/validation-functions.php:357]:
```suggestion
function validate_not_empty( $value, string $message ): void {
```

File: includes/api/validation-functions.php [hunk 3]
```diff
@@ -359,7 +373,9 @@ function validate_not_empty( $value, string $message = 'The field cannot be empt
  *
  * @return void
  */
-function validate_min( $value, int $min, string $message = 'The field must be at least the minimum' ): void {
+function validate_min( $value, int $min, string $message = null ): void {
```
Comment (mindctrl) [inline includes/api/validation-functions.php:376]:
```suggestion
function validate_min( $value, int $min, string $message ): void {
```

**Review by mindctrl — CHANGES_REQUESTED** [review]
Looks good. Some suggestions for consistency.

**Comment by anthonyburchell** [comment]
@mindctrl seems it is failing tests now because that function expects message to be defined

**Comment by anthonyburchell** [comment]
@mindctrl should be good now! Tests passing ✅

**Review by rfmeier — APPROVED** [review]
Good catch on changing `??` (php 7.0+) to `?:` (php 5.3+)

**Review by mindctrl — APPROVED** [review]

**Pull request merged**